File: pages/login.page.ts
import { expect, type Locator, type Page } from '@playwright/test';
import { BasePage } from './base.page';

export class LoginPage extends BasePage {
	private readonly loginField: Locator;
	private readonly passwordField: Locator;
	private readonly logInBtn: Locator;
	private readonly _failedLogInInfo: Locator;

	constructor(page: Page) {
		super(page);
		this.loginField = page.locator("xpath=.//*[@id='field_email']");
		this.passwordField = page.locator("xpath=.//*[@id='field_password']");
		this.logInBtn = page.locator(
			"xpath=.//*[@class='login-form-actions']/input"
		);
		this._failedLogInInfo = page.locator(
			"xpath=.//*[contains(@class,'login_error')]"
		);
		this.checkPage();
	}
	public async setLogin(login: string) {
		await this.loginField.fill(login);
		return this;
	}

	public async setPassword(password: string) {
		await this.passwordField.fill(password);
		return this;
	}

	public async clickLogInBtn() {
		await this.logInBtn.click();
	}

	public get failedLogInInfo(): Locator {
		return this._failedLogInInfo;
	}

	async checkPage() {
		await expect(this.loginField.isEnabled()).toBeTruthy();
		await expect(this.passwordField.isEnabled()).toBeTruthy();
		await expect(this.logInBtn.isEnabled()).toBeTruthy();
	}
}

File: src/test/java/Pages/LoginPage.java
package Pages;

import com.microsoft.playwright.Locator;
import com.microsoft.playwright.Page;


public class LoginPage {
    private final Page page;
    private static final String loginInput = "//*[@id=\"field_email\"]";
    private static final String passwordInput = "//*[@id=\"field_password\"]";
    private static final String loginButton = "Log in to OK";


    public LoginPage(Page page) {
        this.page = page;
        checkLoginPage();
    }

    public void signIn(String login, String password){
        page.locator(loginInput).fill(login);
        page.locator(passwordInput).fill(password);
        page.getByText(loginButton).click();

    }
    public void checkLoginPage(){
        page.locator(loginInput).isVisible();
        page.locator(passwordInput).isVisible();
        page.locator(loginButton).isVisible();
    }


}
